### DISCUSSION TITLE
Distributing a textual application cross platform (`.exe`, `.dmg`, `.deb`)

### DISCUSSION THREAD

**Discussion by JoeZiminski** [Textualize/textual#3834] [Q&A] [answered]
We are interested in distributing our application as a standalone tool for user convenience. This would entail bundling our application with Pyinstaller along with a terminal (or multiple terminals, one for each platform) that support proper rendering of `textual`.  I have used Pyinstaller to bundle software (Qt-based) in `.exe` and `.dmg` before but have never had to consider the terminal. 

I was wondering if this has come up before or anyone had experience with this? I am sure some possible approaches will become apparent after some time playing around but was wondering if this was something that is standard / already optimised? I could not find much googling it.

**Comment by davep** [comment]
There's no "standard" approach to this right now; but I do remember a couple of people talking about approaches they've taken, on our Discord server -- although I seem to remember them being a little more platform-specific.

I think @lllama has done some experimenting around this for example.

**Reply by lllama** [reply]
Have a look at https://github.com/lllama/textualitty 

That's the closest I've got. Currently only works for macOS but the approach should work with Windows and Linux without too much trouble

**Reply by JoeZiminski** [reply, marked as answer]
Thanks a lot that's really useful!

**Reply by BrooksPettit** [reply]
I'm curious if you have tried anything on Windows? I have made an .exe with pyinstaller which works locally, but when I try to run it on an old Windows machine (pre ANSI escape support), it breaks. I was under the impression that textual supports the old WIN32 API. I am wondering if it is an issue with my trying to cross-compile it from Windows 10.

**Comment by KRRT7** [comment]
you can also take a look at Nuitka: https://github.com/Nuitka/Nuitka/
